test-automation-allure/src/main/kotlin/com/nortal/test/report/allure/services/AllureReportZipService.kt

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2022 Nortal AS
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.nortal.test.report.allure.services
+
+import org.springframework.stereotype.Component
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
+
+/**
+ * Compress Allure report to zip file.
+ */
+@Component
+class AllureReportZipService {
+
+    @Throws(IOException::class)
+    fun zipReport(reportDir: String, zipOutputDir: String) {
+        val reportToZip = File(reportDir)
+
+        val zipName = String.format("%s.zip", reportToZip.name)
+        val zipFile = File(zipOutputDir, zipName)
+        zipFile.parentFile.mkdirs()
+
+        FileOutputStream(zipFile).use { fos ->
+            ZipOutputStream(fos).use { zipOut ->
+                zipFile(reportToZip, reportToZip.name, zipOut)
+            }
+        }
+    }
+
+    @Throws(IOException::class)
+    private fun zipFile(fileToZip: File, fileName: String, zipOut: ZipOutputStream) {
+        if (fileToZip.isHidden) {
+            return
+        }
+        if (fileToZip.isDirectory) {
+            if (fileName.endsWith("/")) {
+                zipOut.putNextEntry(ZipEntry(fileName))
+                zipOut.closeEntry()
+            } else {
+                zipOut.putNextEntry(ZipEntry("$fileName/"))
+                zipOut.closeEntry()
+            }
+            val children = fileToZip.listFiles()
+            for (childFile in children!!) {
+                zipFile(childFile, fileName + "/" + childFile.name, zipOut)
+            }
+            return
+        }
+        val fis = FileInputStream(fileToZip)
```

Comment:
Another stream that can be refactored in same manner

Reply:
Refactored